packages/malloy/src/model/malloy_query.ts

```diff
@@ -4429,10 +4449,20 @@ class QueryStruct extends QueryNode {
         return '';
       case 'query': {
         // cache derived table.
-        const name = getIdentifier(this.fieldDef);
+        const clonedAnnotation = structuredClone(
+          this.fieldDef.structSource.query.annotation
+        );
+
+        if (clonedAnnotation) {
+          clonedAnnotation.inherits = undefined;
+        }
+
+        const sourceTag = Tag.annotationToTag(clonedAnnotation).tag;
+
+        const name = this.fieldDef.structSource.query.name;
         // this is a hack for now.  Need some way to denote this table
         //  should be cached.
-        if (name.includes('cache')) {
+        if (name?.includes('cache')) {
```

Comment:
I think we can remove this behavior. AFAIK it is not used anywhere and nothing integrates with it.